packages/react-divider/src/components/Divider/useDividerStyles.ts

```diff
@@ -1,428 +1,240 @@
+    },
+  },
+  horizontalEnd: {
+    ':before': {
+      marginRight: 'var(--divider-borderMargin)',
+    },
+    ':after': {
+      maxWidth: '8px',
+      content: '""',
+      marginLeft: 'var(--divider-borderMargin)',
+    },
+  },
+  horizontalCenter: {
+    ':before': {
+      marginRight: 'var(--divider-borderMargin)',
+    },
+    ':after': {
+      marginLeft: 'var(--divider-borderMargin)',
+    },
+  },
+  important: {
+    '--divider-fontWeight': `700`,
+  },
+  verticalColored: {
+    ':before': {
+      borderRightColor: 'var(--divider-color)',
+    },
+    ':after': {
+      borderRightColor: 'var(--divider-color)',
+    },
+  },
+  horizontalColored: {
+    ':before': {
+      borderTopColor: 'var(--divider-color)',
+    },
+    ':after': {
+      borderTopColor: 'var(--divider-color)',
+    },
+  },
+  verticalChildless: {
+    ':before': {
+      marginBottom: 0,
+    },
+  },
+  inset: {
+    paddingLeft: '12px',
+    paddingRight: '12px',
+  },
+  verticalInset: {
+    marginTop: '12px',
+    marginBottom: '12px',
+  },
+});
 
 /** Applies style classnames to slots */
-export const useDividerStyles = (state: DividerState) => {
-  const rootClassName = useRootStyles(state);
-
-  state.className = ax(rootClassName, state.className);
+export const useDividerStyles = (s: DividerState) => {
+  const styles = useStylesOverride();
+  s.className = ax(
+    styles.root,
+    !s.children && styles.childless,
+    s.appearance === 'subtle' && styles.subtle,
+    s.appearance === 'brand' && styles.brand,
+    s.appearance === 'strong' && styles.strong,
+    !s.vertical && styles.horizontal,
+    s.vertical && styles.vertical,
+    s.vertical && s.children !== undefined && styles.verticalWithChildren,
+    s.alignContent === 'start' && styles.start,
+    s.alignContent === 'end' && styles.end,
+    (s.alignContent === 'center' || !s.alignContent) && s.children !== undefined && s.vertical && styles.center,
+    (s.alignContent === 'center' || !s.alignContent) && s.children !== undefined && !s.vertical && styles.center,
+    (s.alignContent === 'center' || !s.alignContent) && s.children === undefined && styles.centerWithoutContent,
+    s.vertical && s.alignContent === 'start' && styles.verticalStart,
+    s.vertical && s.alignContent === 'end' && styles.verticalEnd,
+    s.vertical && (s.alignContent === 'center' || !s.alignContent) && styles.verticalCenter,
+    !s.vertical && s.alignContent === 'start' && styles.horizontalStart,
+    !s.vertical && s.alignContent === 'end' && styles.horizontalEnd,
+    !s.vertical && (s.alignContent === 'center' || !s.alignContent) && styles.horizontalCenter,
+    s.important && styles.important,
+    s.color !== undefined && s.vertical && styles.verticalColored,
+    s.color !== undefined && !s.vertical && styles.horizontalColored,
+    s.children === undefined && s.vertical && styles.verticalChildless,
+    s.inset && !s.vertical && styles.inset,
+    s.inset && s.vertical && styles.verticalInset,
```

Comment:
```suggestion
    s.inset && (s.vertical ? styles.verticalInset : styles.inset),
```

_I suggest to avoid negative conditions as much as possible, especially in mutable exclusive cases_